tgbot/src/api.rs

```diff
@@ -62,111 +72,267 @@ where
 /// Telegram Bot API client
 #[derive(Clone)]
 pub struct Api {
-    executor: Arc<Box<dyn Executor>>,
+    client: Arc<Client>,
     host: String,
     token: String,
 }
 
 impl Api {
     /// Creates a API instance with a given configuration.
-    pub fn new<C: Into<Config>>(config: C) -> Result<Self, Error> {
+    pub fn new<C: Into<Config>>(config: C) -> Result<Self, ApiError> {
         let config = config.into();
+
+        let mut builder = ClientBuilder::new();
+        builder = if let Some(proxy) = config.proxy {
+            builder.proxy(proxy)
+        } else {
+            builder.no_proxy()
```

Comment:
Do we really need it here?

Reply:
If we have not set proxy, we must be sure that proxy is not used